### ISSUE TITLE
[Hue Emulation] Dimmers are set to (requestedLevel - 1)

### ISSUE THREAD

**Issue by openhab-5iver** [openhab/openhab2-addons#3171]
## Steps to Reproduce (for Bugs)
When setting light levels through Alexa using Echo Dots, my lights are being set to the requested level minus one. When setting the light to 1, the light turns off. Here is a log from asking for the lights to be set to 55...
```
2018-01-25 23:13:56.260 [DEBUG] [enhab.io.hueemulation.internal.HueEmulationServlet] - HuState [on: false bri: 140 hue: 0 sat: 0 xy: {0.0 0.0 } ct: 500 alert: none effect: none colormode: ct reachable: true
2018-01-25 23:13:56.261 [DEBUG] [enhab.io.hueemulation.internal.HueEmulationServlet] - HSBType 0,0,54
2018-01-25 23:13:56.262 [DEBUG] [enhab.io.hueemulation.internal.HueEmulationServlet] - sending 54 to DS_Hallway_Dimmer
2018-01-25 23:13:56.267 [INFO ] [smarthome.event.ItemCommandEvent                  ] - Item 'DS_Hallway_Dimmer' received command 54
```

## Possible Solution
I noticed this while testing a fix for [another issue](https://github.com/openhab/org.openhab.binding.zigbee/issues/108). It took me a minute to figure out I'd found a surprisingly similar new issue in a completely different binding. Maybe some of the changes he committed would be useful?

## Your Environment
OH 2.3.0 snapshot 1197
Hue Emulation Service 2.3.0.201801252322

**Comment by kaikreuzer** [comment]
https://github.com/openhab/openhab2-addons/pull/4216 claimed to have fixed it - please comment there, if this isn't the case.

**Comment by gdgeist** [comment]
> @gdgeist What is your proposed fix? If possible, could you prepare a PR?

@davidgraeff I can take a stab at it.  Tone is very hard to convey in these comments but your work here (along with all the contributors) is greatly appreciated!

**Comment by davidgraeff** [comment]
Actually I cannot confirm the described behaviour. This small java test program is simulating the formula in question with a `double` and an `int` as input. With `Math.floor` and without it:

```
    static int MAX_BRI = 254;

    public static void main(String[] args) {
        double v1 = 1;
        int i = (int) Math.floor(v1 * MAX_BRI / 100.0);
        int j = ((int) v1) * MAX_BRI / 100;
        System.out.println(i);
        System.out.println(j);

        int v2 = 1;
        i = (int) Math.floor(v2 * MAX_BRI / 100.0);
        j = v2 * MAX_BRI / 100;
        System.out.println(i);
        System.out.println(j);
    }
```

And the output is always:
```
2
2
2
2
```

**Comment by gdgeist** [comment]
@davidgraeff headed off to work and will try to look at the code tonight (promise) but specific real world behavior on this for setting a dimmer item via the hue emulation from data gathering:
newsetting <=10 then effectivesetting = newsetting -1
newsetting >10 then effectivesetting = newsetting

Sorry wasnt clear earlier - since this is behavior that changes based upon a value threshold - would assume its because of a mapping of a non integer result to an integer and the inherent mismatch between two scales (0-254 and 1-100)


If I am thinking correctly  - the code needs the calculation to be made using 1-255 then when executed mapped to 0-254 but I'll look tonight

**Comment by openhab-5iver** [comment]
@davidgraeff , this one needs to be reopened, but #4401 should correct it.

**Comment by davidgraeff** [comment]
@kaikreuzer I think you misread the comment. It was from the author of the pr that you have merged and the issue should be resolved now. Did I miss anything?

**Comment by kaikreuzer** [comment]
No, I reopened the issue upon request and then merged #4401 under the assumption that Github auto-close will close this issue again. Seems the "fixes..." comment was missing there, so let's manually close it again.
